lsp-completion.el

```diff
@@ -470,12 +470,15 @@ The MARKERS and PREFIX value will be attached to each candidate."
           ;; boundaries
           ((equal (car-safe action) 'boundaries) nil)
           ;; try-completion
-          ((null action) (cl-first (member probe (funcall candidates))))
+          ((null action)
+           (when-let ((cands (funcall candidates)))
```

Comment:
nit: can you use `(not (cl-rest collection))` instead of comparing lenght to 1. I know that we have that on a few places but it is no the correct elisp.

Reply:
@yyoncho as a side note, `cl-rest` or `cdr`?

Reply:
Since `cl-rest` is alias of `cdr` and more expressive at the same time, IMO `cl-rest` is a clear winner here.

Reply:
Well, both convey the same meaning. To me, `cdr` is more intuitive, IMHO.

Reply:
I am in favour of cl-rest (or even rest if it was not deprecated). car/cdr/etc are archaisms kept only for historical reasons and IMHO they should be deprecated even in core code.